web/src/narrow_state.ts

```diff
@@ -1,37 +1,26 @@
 import * as blueslip from "./blueslip";
 import {Filter} from "./filter";
 import type {Term} from "./filter";
-import * as inbox_util from "./inbox_util";
+import * as message_lists from "./message_lists";
 import {page_params} from "./page_params";
 import * as people from "./people";
-import * as recent_view_util from "./recent_view_util";
 import * as stream_data from "./stream_data";
 import type {StreamSubscription} from "./sub_store";
 import * as unread from "./unread";
 
-let current_filter: Filter | undefined;
-
 export let has_shown_message_list_view = false;
 
-export function reset_current_filter(): void {
-    current_filter = undefined;
-}
-
-export function set_current_filter(filter: Filter | undefined): void {
-    current_filter = filter;
-}
-
-export function active(): boolean {
-    return current_filter !== undefined;
-}
-
 export function filter(): Filter | undefined {
-    // Both, `All messages` and
-    // `Recent Conversations` have `current_filter=undefined`
-    return current_filter;
+    // Both, `All messages` and `Inbox` have `undefined` current message list.
+    if (!message_lists.current) {
+        return undefined;
+    }
+
+    return message_lists.current.data.filter;
```

Comment:
Hmm, interesting.

If I'm correct that you're not changing any behavior, just the `narrow_state` APIs, I think this could be a good cluster of refactors to try to do as its own preparatory PR / series of commits (without making material changes to `narrow.js` itself)?

Reply:
We might need some changes to make `message_lists.current === undefined` a valid/supported state as part of that; `all_rendered_message_lists` has an assertion expecting that it's only possible during early loading today.

Reply:
> If I'm correct that you're not changing any behavior, just the narrow_state APIs, I think this could be a good cluster of refactors to try to do as its own preparatory PR / series of commits (without making material changes to narrow.js itself)?

I was actually thinking it doesn't make sense for `All messages` to have `undefined` filter in the world where it is just another narrow.

But, I guess this would feel much more clear in a separate commit than lumped together with this set of changes so that it is easy to test for any bugs when we merge and activate and deactivate functions.

Reply:
Yeah, exactly, we definitely need to change that return value but separable checkpoints are helpful.